### PR TITLE
fix: test this rule ext recording icon

### PR DESCRIPTION
**Pull request by rohanmathur91**
Closes issue: 

## ✅ Checklist:

- [ ] Make sure linting and unit tests pass.
- [ ] No install/build warnings introduced.
- [ ] Verified UI in browser.
- [ ] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

File: browser-extension/mv2/src/background/background.js [hunk 1]
```diff
@@ -1383,7 +1385,7 @@ BG.Methods.startRecordingExplicitly = (tabId, showWidget = true) => {
     return;
   }
 
-  const sessionRecordingData = { explicit: true, showWidget };
+  const sessionRecordingData = { explicit: true, showWidget, markRecordingIcon: true };
```
Comment (nafees87n) [inline browser-extension/mv2/src/background/background.js:1388]:
```suggestion
  const sessionRecordingData = { explicit: true, showWidget};
```

File: browser-extension/mv2/src/client/js/sessionRecorder.js [hunk 2]
```diff
@@ -54,7 +55,16 @@ RQ.SessionRecorder.setup = () => {
...
-  const { config, previousSession, notify, explicit = false, widgetPosition, showWidget, recordingStartTime } = options;
+  const {
+    config,
+    previousSession,
+    notify,
+    explicit = false,
+    widgetPosition,
+    showWidget,
+    recordingStartTime,
+    markRecordingIcon = false,
```
Comment (nafees87n) [inline browser-extension/mv2/src/client/js/sessionRecorder.js:66]:
```suggestion
    markRecordingIcon = true,
```
We should keep it on by default

File: browser-extension/mv2/src/client/js/sessionRecorder.js [hunk 3]
```diff
@@ -114,11 +125,12 @@ RQ.SessionRecorder.addMessageListeners = () => {
     } else if (event.data.action === "sessionRecordingStarted") {
       RQ.SessionRecorder.isRecording = true;
 
-      if (RQ.SessionRecorder.showWidget) {
-        chrome.runtime.sendMessage({
-          action: RQ.CLIENT_MESSAGES.NOTIFY_SESSION_RECORDING_STARTED,
-        });
+      chrome.runtime.sendMessage({
+        action: RQ.CLIENT_MESSAGES.NOTIFY_SESSION_RECORDING_STARTED,
+        markRecordingIcon: RQ.SessionRecorder.showWidget || RQ.SessionRecorder.markRecordingIcon,
```
Comment (nafees87n) [inline browser-extension/mv2/src/client/js/sessionRecorder.js:130]: The function of `showWidget` is different, shouldn't mix it with marking recording icon
```suggestion
        markRecordingIcon: RQ.SessionRecorder.markRecordingIcon,
```
Then `handleSessionRecordingOnClientPageLoad` in background should update as : 
```
if (sessionRecordingConfig?.autoRecording?.mode === "allPages") {
        sessionRecordingData.showWidget = false;
        sessionRecordingData.markRecordingIcon = false;
      }else{
        sessionRecordingData.showWidget = true;
      }
```

**Review by nafees87n — CHANGES_REQUESTED** [review]
In `browser-extension/common/src/custom-elements/session-recording-widgets/manual-mode-widget/index.ts`

The type of `#recordingTimerIntervalId` should be `Node.Timeout | null`

**Comment by rohanmathur91** [comment]
> Node.Timeout

**Review by nafees87n — APPROVED** [review]

**Pull request merged**